src/olympia/versions/models.py

```diff
@@ -142,8 +142,8 @@ def from_upload(cls, upload, addon, platforms, channel, send_signal=True,
             parsed_data = utils.parse_addon(upload, addon)
         license_id = None
         if channel == amo.RELEASE_CHANNEL_LISTED:
-            previous_version = addon.find_latest_version_including_rejected(
-                channel=channel)
+            previous_version = addon.find_latest_version(
+                channel=channel, exclude=())
```

Comment:
In theory, a beta version can only be submitted after a non-beta version -which must have had a license - so it will get a license copied from that... so... I guess it's okay.  (You can argue about the rights and wrongs of skipping the details step for beta versions in an issue with Product 😛 )